modules/social_features/social_group/modules/social_group_request/src/Plugin/ActivityContext/ApprovedRequestJoinGroupActivityContext.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Drupal\social_group_request\Plugin\ActivityContext;
+
+use Drupal\activity_creator\ActivityFactory;
+use Drupal\activity_creator\Plugin\ActivityContextBase;
+use Drupal\social_group_request\Plugin\GroupContentEnabler\GroupMembershipRequest;
+
+/**
+ * Provides a 'ApprovedRequestJoinGroupActivityContext' activity context.
+ *
+ * @ActivityContext(
+ *  id = "approved_request_join_group_activity_context",
+ *  label = @Translation("Approved request to join a group activity context"),
+ * )
+ */
+class ApprovedRequestJoinGroupActivityContext extends ActivityContextBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRecipients(array $data, $last_uid, $limit) {
+    $recipients = [];
+
+    if (!empty($data['related_object'])) {
+      $referenced_entity = ActivityFactory::getActivityRelatedEntity($data);
+
+      /** @var \Drupal\group\Entity\GroupContentInterface $group_content */
+      $group_content = $this->entityTypeManager->getStorage('group_content')
+        ->load($referenced_entity['target_id']);
+
+      if (
+        !$group_content->get('entity_id')->isEmpty() &&
+        $group_content->hasField('grequest_status') &&
+        !$group_content->get('grequest_status')->isEmpty() &&
+        $group_content->get('grequest_status')->value == GroupMembershipRequest::REQUEST_ACCEPTED
+      ) {
+        $recipients[] = [
+          'target_type' => 'user',
+          'target_id' => $group_content->entity_id->target_id,
```

Comment:
Sorry for this drive-by comment 😅 

@OH17L If you really only need the ID of the entity then I it's probably fine to just read the `target_id` of the field. I believe (but verification is needed) that calling `getEntity` will load the entire entity. So this could be a significant performance penalty over simply moving a number around.